js/components/Modal.tsx

```diff
@@ -126,11 +126,14 @@ export const ModalCloseButton = ({ onClose }: ModalCloseButtonProp) => {
     );
 };
 
+type OnCloseCallback = (closeBox?: boolean) => void;
+
 /**
  * ModalHeader component properties
  */
+
 export interface ModalHeaderProps {
-    onClose?: (closeBox?: boolean) => void;
+    onClose?: OnCloseCallback | undefined;
```

Comment:
Again, I don't know why to also say that `onClose` could be explicitly `undefined`.

Reply:
This is happening because I'm explicitly referencing onClose in the containing component, so it because no longer optional from that perspective, but since it was optional to be passed in, it can be undefined. I can work around this with spread operators, possibly.